packages/block-library/src/query/edit/query-placeholder.js

```diff
@@ -122,6 +122,7 @@ function QueryVariationPicker( {
 						setAttributes( {
 							...variation.attributes,
 							query: {
+								...attributes.query,
 								...variation.attributes.query,
 								postType:
```

Comment:
I actually don't understand why we're doing this, Why are we overriding namespace and post type here?

Reply:
I think we should just call `setAttributes( variation.attributes )` since this variation picked is only used when starting blank.

Reply:
>I think we should just call setAttributes( variation.attributes )

Hm.. that could work now. If I remember correctly, this was needed because before we intentionally had the `inherit:false` and since properties of  object attributes are not merged, we needed to also provide the rest of the props in the variation declaration. But since we could be here from another `inserter` Query variation, we needed to respect some important attributes like `postType`. I'll update to just set the attributes then.

Reply:
Ah.. it's also about this PR: https://github.com/WordPress/gutenberg/pull/46410. `namespace` is used in order to be able to connect variations with variations 😄 .  So we cannot do just `setAttributes`..

Reply:
> namespace is used in order to be able to connect variations with variations 😄 . So we cannot do just setAttributes..

What? I don't understand this sentence, can you clarify?

Reply:
What is "namespace" attribute and why do we need it?

Reply:
Can you give a quick read of [this PR's](https://github.com/WordPress/gutenberg/pull/46410) description? That would be it.

In general the `namespace` attribute is something like an identifier for Query Loop in order to help with active variations detection. It's mostly a result of `query` being an object attribute and nuances that for Query Loop extenders register variations and there is the need to have a way to detect their variation for various things (control additions, etc..) in both client and server side.

Reply:
Ok, so one thing is: I'm not sure we should have a `namespace` attribute. 

But still, I can't figure out how it relates to this PR. We're starting blank, we're picking variations from a predefined list that are scoped to "block", we should just the attribute of the selected variation.

Reply:
I've read the related PR but I can't figure out why we show the "start blank" option at all for these query variations. IF you provide a variation for the query block, you don't need to show the "block scoped variations" at all, you're not starting blank as you picked a variation no?

Reply:
I'm understand the issue more, I think combining "block variations" with "inserter variations" was a mistake from the start. Do we know of people actually using that or is it theoretical?

Reply:
Here's another proposal: "block variations" shouldn't update the query attribute at all, they should just update the inner blocks. So what about removing the setAttributes here entirely?

Reply:
>IF you provide a variation for the query block, you don't need to show the "block scoped variations" at all, you're not starting blank as you picked a variation no?

Not necessarily. You can register variations that don't provide inner blocks.

>I'm understand the issue more, I think combining "block variations" with "inserter variations" was a mistake from the start. Do we know of people actually using that or is it theoretical?



There were many issues from extenders and was definitely used. Not sure if some plugins still use it, but there was a lot of traction to get this working.

>Here's another proposal: "block variations" shouldn't update the query attribute at all, they should just update the inner blocks. So what about removing the setAttributes here entirely?

This should be a breaking change, no?

Reply:
> This should be a breaking change, no?

I don't think so. For me, it just clarifies the behavior. "Inserter" is about defining the whole block, "block" scoped variations is about the content (which is already the case btw)